src/app/server/Mdns.cpp

```diff
@@ -64,6 +65,26 @@ NodeId GetCurrentNodeId()
     return chip::kTestDeviceNodeId;
 }
 
+// Requires an 8-byte mac to accommodate thread.
+chip::ByteSpan FillMAC(uint8_t mac[8])
+{
+    memset(mac, 0, 8);
+    if (DeviceLayer::ConfigurationMgr().GetPrimary802154MACAddress(mac) == CHIP_NO_ERROR)
+    {
+        ChipLogDetail(Discovery, "Using Thread MAC for hostname.");
+        return chip::ByteSpan(mac, 8);
+    }
+    if (DeviceLayer::ConfigurationMgr().GetPrimaryWiFiMACAddress(mac) == CHIP_NO_ERROR)
+    {
+        ChipLogDetail(Discovery, "Using wifi MAC for hostname");
+        return chip::ByteSpan(mac, 6);
+    }
+    ChipLogError(Discovery, "Wifi mac not known. Using a default.");
```

Comment:
What about Ethernet devices, we have some that folks are testing with?

Reply:
That is a good question - Spec specifies mac for ethernet too, but it's probably in a different config. I'll take a look and add that in. Thanks.

That does bring up another interesting question though - for dual devices, what should we prefer? With the code as it is right now, border routers will prefer the EUI-64, which may not actually be what we want. Ditto for dual ethernet / wifi devices - if the ethernet mac comes from a different config, we may want a way to test what's actually connected. Currently, it's just an ID so it doesn't actually matter too much for functionality, and the spec doesn't have a stated preference.